### PR TITLE
DDBCMS-371 - Subsearch: Various fixes v2.

### PR DESCRIPTION
**Pull request by bozeak**
https://inlead.atlassian.net/browse/DDBCMS-371

- Show "Secondary search" only if there is data to be shown.
- Fix display of "Secondary search" block if is set position "After".
- Correction of "ratio" calculations for more precision.

### CODE REVIEW

File: ting_subsearch_secondary.module [hunk 1]
```diff
@@ -130,18 +146,18 @@ function ting_subsearch_secondary_search($keys, $results) {
...
       }
 
-      $nonfiction = 1;
+      $nonfiction = (int) 1;
       if (!empty($genre_facets['nonfiktion'])) {
         $nonfiction = $genre_facets['nonfiktion']->getCount();
       }
 
-      $fiction = 0;
+      $fiction = (int) 0;
```
Comment (OIOIOOOI) [inline ting_subsearch_secondary.module]: Any reason to explicitly cast the type? It's anyway an `int`, unless it's `1.0f`.

**Comment by OIOIOOOI** [comment]
Would suggest to run a code sniffing tool through the module.

File: ting_subsearch_secondary.module [hunk 2]
```diff
@@ -176,7 +192,7 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
   if ($secondary_num_objects > $secondary_result_min && $secondary_num_objects < $secondary_max) {
     $ratio = $secondary_num_objects / $num_total_objects;
 
-    if ($ratio > variable_get('ting_subsearch_secondary_ps_factor', 2)) {
+    if ($ratio > ting_subsearch_common_tofloat(variable_get('ting_subsearch_secondary_ps_factor', 2))) {
```
Comment (OIOIOOOI) [inline ting_subsearch_secondary.module]: All of these `ting_subsearch_common_tofloat` seems like a hacky way of manipulating data in background.
I assume these variable settings come from some sort of admin interface, so I would suggest to simply create a form validation mechanism for fields that might accept floating point values.
And a `hook_update` would transform existing ones to proper format.

Comment (bozeak) [reply ting_subsearch_secondary.module]: Added `hook_update` and custom form element validation in `ting_subsearch_common` to ensure that the correct float values separator is used.

**Review by OIOIOOOI — APPROVED** [review]

**Pull request merged**